components/webvr/webvr_thread.rs

```diff
@@ -0,0 +1,349 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use ipc_channel::ipc;
+use ipc_channel::ipc::{IpcReceiver, IpcSender};
+use msg::constellation_msg::PipelineId;
+use script_traits::{ConstellationMsg, WebVREventMsg};
+use std::{thread, time};
+use std::collections::{HashMap, HashSet};
+use std::sync::mpsc;
+use std::sync::mpsc::{Receiver, Sender};
+use webrender_traits;
+use webvr_traits::{WebVRMsg, WebVRResult};
+use webvr_traits::webvr::*;
+
+/// Defines the polling interval time in ms for VR Events such as VRDisplay connected, disconnected, etc.
+const EVENT_POLLING_INTERVAL: u64 = 500;
```

Comment:
Good question. There isn't a formula to decide the perfect value, but 500 seemed like a good balance between resource usage and desired precision. I saw similar values in native games and other implementations. In general these events (ie discovering new displays) don't need very high precision. We can reduce the value if needed because the pooling thread is only active when there is an active tab using WebVR APIs.